### PR TITLE
AG-8932 - Add preliminary charts option metadata generation

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-8932

Initial support for `ag-charts-community` generation of `interfaces.AUTO.json` and `doc-interfaces.AUTO.json` in the form of an Nx executor.

### CODE REVIEW

File: plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/formatAST.cjs
Comment (alantreadway) [inline plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/formatAST.cjs:1]: This file is taken from: https://github.com/ag-grid/ag-grid/blob/latest/grid-community-modules/all-modules/generate-code-reference-files.js

File: plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/executor.ts
Comment (alantreadway) [inline plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/executor.ts:1]: This file is taken from: https://github.com/ag-grid/ag-grid/blob/latest/scripts/formatAST.js

**Review by taktran — APPROVED** [review]
👍🏼

File: plugins/ag-charts-build-tools/src/index.ts
Comment (manapeirov) [inline plugins/ag-charts-build-tools/src/index.ts:1]: Is this empty file supposed to be here?

**Review by manapeirov — APPROVED** [review]

**Pull request merged**